src/services/__tests__/fhrsApi.test.ts

```diff
@@ -0,0 +1,161 @@
+import { getEstablishmentRatingValue } from '../fhrsApi';
+
+describe('fetchRating', () => {
+  it('should return a rating between 0 and 5', () => {
+    const fhrsIds = [
+      '268458',
+      '269382',
+      '268811',
+      '268903',
+      '269258',
+      '269026',
+      '269099',
+      '269095',
+      '809473',
+      '268631',
+      '269312',
+      '269253',
+      '988122',
+      '723651',
+      '269229',
+      '268672',
+      '268669',
+      '269078',
+      '854391',
+      '708562',
+      '268508',
+      '989695',
+      '268628',
+      '268629',
+      '269386',
+      '268937',
+      '269058',
+      '268487',
+      '1065520',
+      '268323',
+      '268324',
+      '268322',
+      '842962',
+      '268874',
+      '269309',
+      '269148',
+      '269171',
+      '268488',
+      '777803',
+      '268630',
+      '269224',
+      '268617',
+      '268615',
+      '268618',
+      '269431',
+      '269384',
+      '269383',
+      '999835',
+      '268752',
+      '269248',
+      '269495',
+      '269497',
+      '268490',
+      '977892',
+      '268807',
+      '729281',
+      '269464',
+      '269158',
+      '848733',
+      '268780',
+      '952041',
+      '268481',
+      '269028',
+      '949726',
+      '269039',
+      '542347',
+      '1015463',
+      '268624',
+      '269252',
+      '268982',
+      '679551',
+      '949727',
+      '989697',
+      '406128',
+      '997133',
+      '836583',
+      '997140',
+      '743833',
+      '988109',
+      '268454',
+      '964599',
+      '1011214',
+      '269163',
+      '841948',
+      '889450',
+      '268806',
+      '1015461',
+      '268453',
+      '929914',
+      '269262',
+      '268351',
+      '269202',
+      '268849',
+      '777807',
+      '268441',
+      '997130',
+      '1031734',
+      '661417',
+      '269447',
+      '269434',
+      '269449',
+      '268866',
+      '269448',
+      '269442',
+      '269441',
+      '269436',
+      '269437',
+      '269446',
+      '269438',
+      '269452',
+      '269433',
+      '955351',
+      '269493',
+      '268355',
+      '269260',
+      '269361',
+      '268548',
+      '268352',
+      '269402',
+      '681923',
+      '269142',
+      '268353',
+      '269391',
+      '661418',
+      '904891',
+      '268433',
+      '268667',
+      '1401890',
+      '268356',
+      '268993',
+      '268935',
+      '268934',
+      '268933',
+      '268435',
+      '269201',
+      '848730',
+      '268461',
+      '268410',
+      '668853',
+      '268447',
+      '269155',
+      '268450',
+      '1292333',
+    ];
+
+    fhrsIds.forEach(async (id) => {
+      const rating = await getEstablishmentRatingValue(parseInt(id, 10));
```

Comment:
Could you please mock the `fetchJson` to return an example response?
```
jest
      .spyOn(fetch, 'fetchJson')
      .mockResolvedValue(`{ response json ... }`);
```

Also If the response format is usally the same, one example is enough.

We write unit tests to test our conversion functions. Your test would check if API is working correctly, but if it was down it would block our pipeline. Good rule is "unit tests must work offline".

It could be good to know if shape of their API changed, but this is not role of unit tests. We could have perhaps some "3rd party service status page" checking that.

Reply:
So I changed it now, but I am not sure if it is right to have the entire API response or if it would be right to just have the relevant parts of the response.

Reply:
Yes, that is perfect. We call it fixtures, and its fine to have long JSONs in tests. This test isn't that much useful, but at least we know what other info is in that api call - for future reference.

✅